app/src/main/java/org/mozilla/fenix/perf/RunBlockingCounter.kt

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// This class implements the alternative ways to invoke runBlocking with some
+// monitoring by wrapping the raw methods. This lint check tells us not to use the raw
+// methods so we suppress the check.
+@file:Suppress("MozillaRunBlockingCheck")
+
+package org.mozilla.fenix.perf
+
+import androidx.annotation.VisibleForTesting
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.runBlocking
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Counts the number of runBlocking calls made
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+object RunBlockingCounter {
+    var count = 0L
+        @Synchronized set
+}
+
+/**
+ * Wrapper around `runBlocking`. RunBlocking seems to be a "fix-all" to return values to the thread
+ * where the coroutine is called. The official doc explains runBlocking:  "Runs a new coroutine and
+ * blocks the current thread interruptibly until its completion`. This can have negative
+ * side-effects on the our main thread which could lead to significant jank. This wrapper aims to
+ * count the number of runBlocking call to try to limit them as much as possible to encourage
+ * alternatives solutions whenever this function might be needed.
+ */
+fun <T> runBlockingIncrement(
+    context: CoroutineContext? = null,
+    action: suspend CoroutineScope.() -> T
+): T {
+    RunBlockingCounter.count += 1
+
+    context?.let {
```

Comment:
This doesn't do what you think it does. It does:
```kotlin
if (context != null) {
    val returnValue = runBlocking(context) { action() }
    if (returnValue != null) {
        runBlocking(action())
    }
}
```

I guess unless return return from the containing function and not the lambda. Even if the code is correct, the fact that it can be misinterpreted isn't great. In general, it's good to be careful when using symbols (`?:`) or unintuitively short functions (`let`, `run`, and friends) to express something.

I would prefer
```kotlin
return if (context != null) {
    runBlocking(it, ::action)
} else {
    runBlocking(::action)
}
```

Note that the function reference is an optimization that avoids us having to create a new lambda (if the compiler isn't smart enough to do it for us). Note also the deduplication of the `return` statement.